### DISCUSSION TITLE
custom icons font usage

### DISCUSSION THREAD

**Discussion by flamby** [Textualize/rich#1627] [Q&A]
Hi,

I generated a custom icons font thanks to https://icomoon.io/, then installed it on my OS (macOS, font is visible in the font browser).
I naively thought that I would be able to use it right away in rich or textual with the :icon-name: approach (in a Tree object) 
Have I missed something? Or maybe it's not that possible yet.

Thanks and keep the good work!

**Comment by willmcgugan** [comment]
It's a function of your terminal software. If its MacOS terminal there is an option to select your font.

**Reply by flamby** [reply]
Thanks for the clarification. 
I thought that I could use it like in an html page, i.e. by having multiple font-family.
Since my custom font has only icons and no ascii in it, I'm afraid it won't work.
Anyway, I tried changing both ascii and non-ascii fonts of iTerm to my custom font, and it does not display the custom icon, and of course, all the already displayed stuff of the term is broken afterward.

Have you try it with a custom font as well?
Perhaps the ttf way is not a good idea. icomoon generates an svg as well.

**Reply by mrvkino** [reply]
I am not sure if this will help, but you can look at Nerd Font [Font Patcher Script](https://www.nerdfonts.com/) (look at the bottom of the page). They patch font file with custom symbol. 

The only thing is you need to associate a utf-8 code to each symbol so it knows where to put in in the font file.

**Reply by flamby** [reply]
thanks @mrvkino I'll try that and update the discussion, that may help others.
I did not bother associating utf-8 code w/ my icomoon's generated font, that may be part of the issue as well, and thus will retry it too.
